libr/asm/op.c

```diff
@@ -0,0 +1,73 @@
+/* radare - LGPL - Copyright 2018 - pancake */
+
+#include <r_asm.h>
+
+R_API RAsmOp *r_asm_op_new() {
+	return R_NEW0 (RAsmOp);
+}
+
+R_API void r_asm_op_free(RAsmOp *op) {
+	r_asm_op_fini (op);
+	free (op);
+}
+
+R_API void r_asm_op_init(RAsmOp *op) {
+	memset (op, 0, sizeof (*op));
+}
+
+R_API void r_asm_op_fini(RAsmOp *op) {
+	r_strbuf_fini (&op->buf);
+	r_strbuf_fini (&op->buf_asm);
+	r_strbuf_fini (&op->buf_hex);
+	r_buf_fini (op->buf_inc);
+}
+
+// accessors
+R_API char *r_asm_op_get_hex(RAsmOp *op) {
+	return r_strbuf_get (&op->buf_hex);
+}
+
+R_API char *r_asm_op_get_asm(RAsmOp *op) {
+	return r_strbuf_get (&op->buf_asm);
+}
+
+R_API int r_asm_op_get_size(RAsmOp *op) {
+	if (!op) {
+		return 1;
+	}
+	const int len = op->size - op->payload;
+	return R_MAX (1, len);
+}
+
+R_API void r_asm_op_set_asm(RAsmOp *op, const char *str) {
+	r_strbuf_set (&op->buf_asm, str);
+}
+
+R_API void r_asm_op_set_hex(RAsmOp *op, const char *str) {
+	r_strbuf_set (&op->buf_hex, str);
+	ut8 *bin = (ut8*)strdup (str);
+	if (bin) {
```

Comment:
I would check for !bin and return in that case. It makes the code easier to read because of less indentation